### ISSUE TITLE
Moving the WHERE clause to the base query to reduce table scan load

### ISSUE THREAD

**Issue by callum-mcdata** [dbt-labs/dbt_metrics#42]
## To Where Or Not To Where

@joellabes had a great point in our discussion today around the WHERE clause added in #34. He stated that it could potentially break our idea of the metric being the single point of access for centrally approved metric calculations - ie, where clauses could cause different datasets to be returned. 

I'm torn on the subject. In the end I think it comes down to how much flexibility we want to give people.

### Paths Forward

1. We remove the WHERE clause
2. We add the WHERE clause to the **end** of the generated SQL instead of the base query. This would limit functionality to only filtering on either the metric value (ie arbitrarily excluding or including values based on some condition) or the dimensions provided in the metric definitions. 
3. Leave as-is (as raised by @joellabes despite his opposition)

@jasnonaz I know you were a proponent of implementing this functionality and curious to hear your thoughts.

**Comment by drewbanin** [comment]
oh boy - @callum-mcdata - the inclusion of "regex" in this conversation is starting to scare me! I wonder if there's some more core decision we need to navigate before answering this question... Can we sync up before we implement anything like this?

**Comment by joellabes** [comment]
➕💯 - I remain a noted curmudgeon about using regex to introspect queries! 

I saw your original notification and was going to hold off on commenting until I was back next week. Since I'm here though, I'm keen to join that discussion if you need me. (Drew will do a good enough job articulating my opinion if you have to move faster)

**Comment by callum-mcdata** [comment]
😅 I think I scared both of you - I'll confess this is not on the docket for the next few weeks but the idea struck me on a walk which is why I added the comment to get the information written down. Love the 🔕 feedback though!

The only other alternative that comes to mind is to follow the design pattern of other tools in this area and do something like:
```
where = [dimension_name:'value', dimension_name:"value_1, value_2"]
```

I dislike this pattern though because it doesn't feel particularly .... SQL? But then again I'm not a huge fan of our current `filter` implementation **and** our ideal solution works best for integration partners and not sql writers. 🤔
